internal/clients/http/sources/sources_client.go

```diff
@@ -98,8 +101,11 @@ func (c *sourcesClient) ListProvisioningSourcesByProvider(ctx context.Context, p
 		return nil, fmt.Errorf("failed to get provider name according to sources service: %w", err)
 	}
 
-	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddSourcesIdentityHeader,
-		headers.AddEdgeRequestIdHeader, BuildQuery("filter[source_type][name]", sourcesProviderName))
+	limit := strconv.Itoa(ctx.Value(middleware.LimitCtxKey).(int))
+	offset := strconv.Itoa(ctx.Value(middleware.OffsetCtxKey).(int))
```

Comment:
Type assertion like this is dangerous - Value can also return `nil` which will lead to panic. As I suggest below, create a function to do all the hard work of checking for `nil` etc.

Reply:
I will create the function as you suggested, however I don't think that it will ever be nil because in the middleware offset and limit are assigned with default values (0, 100)

Reply:
Can you explain Why should I do the check  in a separate function  and not in the middleware? 
I am always setting a value for offset and limit. 
Do you suggest to transfer  the validation check to the function because of the exported keys?